### DISCUSSION TITLE
[C#] Simplified if-elseif block on some simple bool conditions.

### DISCUSSION THREAD

**Discussion by Flithor** [dotnet/roslyn#60740] [Ideas]
I think the below code can add some quick fix
```
If (boolValue) // or: "value != null" / "value == null"
{
}
//Depending on the code context, "!boolValue" is a nonsense condition, should can be quick fix
else if (!boolValue && anotherCondition)
{
}
//Depending on the code context, "!boolValue" is a nonsense condition, should can be quick fix
else if (!boolValue)
{
}
//Unreachable, but no warning
else if (!boolValue && anyOtherCondition)
{
}
else
{
}
```

**Comment by CyrusNajmabadi** [comment]
Moving to roslyn analyzers

**Reply by Flithor** [reply]
Hey, Alpaca dude, dose it moved?

**Comment by SakuraIsayeki** [comment]
I could argue that depending on edge cases, there are times where reevaluating a property or reexecuting a method in a following if/else statement would make sense. 

Auditing and statistics for one, you can have a small counter being incremented each time an expensive method, for example. 
And let's argue for another one that your extremely expensive property changed state halfway through the if-else statements ; Then at this point, reevaluation would actually be necessary to not fall on assumptions made obsolete by implicitly trusting the first statement in the block.

Imagine: 
```cs
if (getBoolValue()) // This one is querying another running thread, with its own state
{

}
// Depending on the code context, "!getBoolValue()" COULD BE a nonsense condition,
// but any change to the parallel thread borks the result So reevaluation is necessary...
else if (!getBoolValue() && anotherCondition)
{

}
// You start to get my point by now.
else if (!getBoolValue())
{

}
// ...And in these conditions, there are so many unknowns. So just to be sure...
else 
{

}
```

Hope I got my point across clearly enough to make sense ^^

**Reply by Flithor** [reply]
**simple bool conditions** in title.

Obviously, just bool type local variable or structure member can be called a "simple bool condition".

The result of calling method or complex expression are can not inferred definitely.
